boards/arm/beagle_bcf/board.cmake

```diff
@@ -5,5 +5,6 @@
 
 # Download cc2538-bsl.py from https://git.beagleboard.org/beagleconnect/zephyr/cc2538-bsl/-/tags/2.1-bcf
 
+find_program(CC1352_FLASHER NAMES cc1352_flasher)
 board_set_flasher_ifnset(misc-flasher)
-board_finalize_runner_args(misc-flasher $ENV{ZEPHYR_BASE}/boards/arm/beagle_bcf/cc2538-bsl.py -w)
+board_finalize_runner_args(misc-flasher ${CC1352_FLASHER} --bcf)
```

Comment:
What you should have been able to do was to substitute the runner with a message, so that it prints the needed info, for example like this:
```
if(CC1352_FLASHER)
  board_finalize_runner_args(misc-flasher ${CC1352_FLASHER} --bcf)
else()
  board_finalize_runner_args(misc-flasher ${CMAKE_COMMAND} -E echo 
    "cc1352_flasher not installed, please install it using pip install cc1352-flasher"
  )
endif()
```

which will then print the help text when running `west flash`, but still allow the build to pass.

However, for that solution to work, then this line: https://github.com/zephyrproject-rtos/zephyr/blob/d019474766c1f4143db22a960e86c8ecd22c1961/scripts/west_commands/runners/misc.py#L51
must be updated to not pass on `self.cfg.build_dir` to the command.

Something which is quite puzzling to me, because how can we know if a misc runner can handle build dir as argument at this location ?
I assume this even will cause problems to `cc1352_flasher` invocation ?

@mbolivar-ampere Do you have any recollection as to why build dir is always passed as first argument to the misc runner ?
Seems to me that if build dir is to be passed to the flash command, it should be the board.cmake's descision.